mig-api/api.go

```diff
@@ -438,6 +440,37 @@ func getIP(respWriter http.ResponseWriter, request *http.Request) {
 	respond(http.StatusOK, []byte(remotePublicIP(request)), respWriter, request)
 }
 
+// getPublicKey takes an pgp_fingerprint and returns corresponding publickey
+func getPublicKey(respWriter http.ResponseWriter, request *http.Request) {
+	var err error
+	opid := getOpID(request)
+	defer func() {
+		if e := recover(); e != nil {
+			emsg := fmt.Sprintf("%v", e)
+			ctx.Channels.Log <- mig.Log{OpID: opid, Desc: emsg}.Err()
+			respond(http.StatusInternalServerError, emsg, respWriter, request)
+		}
+		ctx.Channels.Log <- mig.Log{OpID: opid, Desc: "leaving getPublicKey()"}.Debug()
+	}()
+	vars := mux.Vars(request)
+	fp := vars["pgp_fingerprint"]
+
+	// retrieve the publickey
+	var inv mig.Investigator
+	if len(fp) > 0 {
```

Comment:
Should we just error out here if fp isn't valid? Right now in this case it looks like we would still call pgp.ArmorPubKey() later with the uninitialized investigator value, since that part of the function will still run.